src/contexts/AuthContext.tsx

```diff
@@ -24,9 +24,13 @@ export default function AuthContextProvider({ children }: PropsWithChildren) {
     if (urlSearchParams.has('key')) {
       const key = urlSearchParams.get('key');
       const body: ApiKeyParameters = { key };
-      setUser(await AuthenticationService.authKey(body));
+      await authKey({ body }).then((res) => {
+        setUser(res.data);
+      });
```

Comment:
Again, I am more fan of `const res = await authKey({ body });` because this is why `await` was created in the first place (to not have these infinite `.then()`s.

Reply:
Went over the files again and removed all `await-then` clauses where possible.